src/main/resources/assets/src/managerView/branchHoursView.js

```diff
@@ -1,13 +1,283 @@
+        const branch = this.props.branches.find(b => b.name == newValues[2]);
+        const dayOfWeek = this.convertDayToInt(newValues[3]);
+
+        const validOpenTime = openTime.match('^(0[0-9]|1[0-9]|2[0-3]|[0-9]):[0-5][0-9]$');
+        const validCloseTime = closeTime.match('^(0[0-9]|1[0-9]|2[0-3]|[0-9]):[0-5][0-9]$');
+
+        // Check if the data is valid
+        if (!validOpenTime || !validCloseTime || !branch || !dayOfWeek) {
+            const newEditorData = this.state.editorData;
+
+            if (!validOpenTime) {
+                newEditorData[0] = "Invalid open time";
+            }
+
+            if (!validCloseTime) {
+                newEditorData[1] = "Invalid close time";
+            }
+
+            if (!branch) {
+                newEditorData[2] = "Invalid name";
+            }
+
+            if(!dayOfWeek){
+                newEditorData[3] = "Invalid day of week";
+            }
+
+            this.setState({
+                editorData: newEditorData,
+            })
+
+            this.props.setStateValue('loading', false);
+            return;
+        }
+
+        const branchId = branch.id;
+
+        if (id == -1) {
+            await $.ajax({
+                type: 'POST',
+                headers: {
+                    'Accept': 'application/json',
+                    'Content-Type': 'application/json'
+                },
+                url: '/api/branchHours/add',
+                data: JSON.stringify({
+                    openTime: openTime,
+                    closeTime: closeTime,
+                    branchId: branchId,
+                    dayOfWeek: dayOfWeek,
+                })
+            });
+        } else {
+            await $.ajax({
+                type: 'POST',
+                headers: {
+                    'Accept': 'application/json',
+                    'Content-Type': 'application/json'
+                },
+                url: '/api/branchHours/update',
+                data: JSON.stringify({
+                    openTime: openTime,
+                    closeTime: closeTime,
+                    branchId: branchId,
+                    dayOfWeek: dayOfWeek,
+                    id: id,
+                })
+            });
+        }
+
+        await this.props.loadData('branchHours');
+        this.props.setStateValue('loading', false);
+
+        this.setState({
+            showEditDialog: false,
+        })
+    }
+
+    closeEditor() {
+        this.setState({
+            showEditDialog: false,
+        })
+    }
+
+    setFilters(newFilters) {
+        this.setState({
+            filters: newFilters,
+        })
+    }
+
+    clearFilters() {
+        this.setState({
+            filters: [[], [], [], []],
+        })
+    }
+
+    getBranchName(id) {
+        return this.props.branches.find(b => { return b.id == id }).name;
+    }
+
+    convertIntToDay(dayNum) {
+        switch(dayNum){
+            case 1:
```

Comment:
I think 1 start with Monday.

Reply:
Oh my bad, I was basing it off of the calendar table's Sunday through Saturday 1-7 format.

Reply:
I know we talked about this but I can't remember what we decided, why are the days as ints for branch hours different from the calendar days?

Reply:
I’m not sure I understand. Branch hours has day of week for weekly hours. Calendar days or id is from 1-365

Reply:
Sorry, I meant in the calendar table there is a column called Day_Of_Week and it uses 1-7 starting with Sunday. I guess the two tables don't really interact with each other so it doesn't matter.

Reply:
We don’t use that column.

Reply:
Exactly, I was more just curious for the sake of uniformity. But it doesn't matter either way.